examples/mnist/run.py

```diff
@@ -57,43 +64,33 @@ def normalize(image, label):
   def _build_training_graph(self, training_data):
     """Build a graph for plaintext model training."""
 
-    # model parameters and initial values
-    w0 = tf.Variable(tf.random_normal([28 * 28, 512]))
-    b0 = tf.Variable(tf.zeros([512]))
-    w1 = tf.Variable(tf.random_normal([512, 10]))
-    b1 = tf.Variable(tf.zeros([10]))
-
-    # optimizer and data pipeline
-    optimizer = tf.train.AdamOptimizer(learning_rate=0.01)
-
-    # training loop
-    def loop_body(i):
-
-      # get next batch
-      x, y = training_data.get_next()
+    model = keras.Sequential()
+    model.add(keras.layers.Flatten())
+    model.add(keras.layers.Dense(512, input_shape=[28, 28, 1]))
+    model.add(keras.layers.Activation('relu'))
+    model.add(keras.layers.Dense(self.NUM_CLASSES, activation='softmax'))
 
-      # model construction
-      layer0 = tf.matmul(x, w0) + b0
-      layer1 = tf.nn.sigmoid(layer0)
-      layer2 = tf.matmul(layer1, w1) + b1
+    def loss(model, x, y):
+      y_ = model(x)
+      return tf.losses.sparse_softmax_cross_entropy(labels=y, logits=y_)
 
-      predictions = layer2
-      loss = tf.reduce_mean(
-          tf.losses.sparse_softmax_cross_entropy(logits=predictions, labels=y))
-      with tf.control_dependencies([optimizer.minimize(loss)]):
-        return i + 1
+    def grad(model, inputs, targets):
+      with tf.GradientTape() as tape:
+        loss_value = loss(model, inputs, targets)
+      return loss_value, tape.gradient(loss_value, model.trainable_variables)
 
-    loop = tf.while_loop(lambda i: i < self.ITERATIONS
-                         * self.EPOCHS, loop_body, (0,))
+    # optimizer and data pipeline
+    optimizer = tf.train.AdamOptimizer(learning_rate=0.01)
+    global_step = tf.Variable(0)
+    for epoch in range(self.EPOCHS):
+      # Training loop - using batches of 32
+      for x, y in training_data.get_next():
```

Comment:
I'm not sure that training_data.get_next() is a generator so I don't think this will work as intended -- I'd revert to something more similar to what we had before:
```python
def loop_body(i):
    x, y = training_data.get_next()
    loss_value, grads = grad(model, x, y)
    update_op = optimizer.apply_gradients(zip(grads, model.trainable_variables), global_step)
    with tf.control_dependencies([update_op]):
        return global_step
```
then you can use the same `tf.while_loop` code we had before, as well as everything after it but returning the `model.trainable_weights` instead of the list